knotx-example/knotx-example-gateway/src/main/java/io/knotx/gateway/impl/RequestProcessorKnotProxyImpl.java

```diff
@@ -53,19 +56,30 @@ protected KnotContext processError(KnotContext knotContext, Throwable error) {
   }
 
   private KnotContext createSuccessResponse(KnotContext inputContext) {
-
     ClientResponse clientResponse = new ClientResponse();
+    String responseBody = getResponseBodyAsString(inputContext);
 
-      io.vertx.rxjava.core.MultiMap headers = clientResponse.getHeaders();
-      headers.add(HttpHeaders.CONTENT_LENGTH.toString().toLowerCase(),
-          Integer.toString(RESPONSE.length()))
-          .add("Content-Type", "application/json");
-
-      clientResponse.setBody(Buffer.buffer(RESPONSE)).setHeaders(headers);
-      clientResponse.setStatusCode(HttpResponseStatus.OK.code());
+    clientResponse.setBody(Buffer.buffer(responseBody))
+        .setHeaders(getHeaders(clientResponse, responseBody));
+    clientResponse.setStatusCode(HttpResponseStatus.OK.code());
 
     return new KnotContext()
         .setClientRequest(inputContext.getClientRequest())
         .setClientResponse(clientResponse);
   }
+
+  private MultiMap getHeaders(ClientResponse clientResponse, String response) {
```

Comment:
So it's more clear if the parameter was simply a body length. currently it suggests you're doing something with the body